content/blog/deploying-a-pern-stack-application-to-aws/index.md

```diff
@@ -0,0 +1,349 @@
+    CREATE SCHEMA voting_app;
+    CREATE TABLE voting_app.choice(
+        choice_id SERIAL PRIMARY KEY,
+        text VARCHAR(255) NOT NULL,
+        vote_count INTEGER NOT NULL
+    );
+    GRANT USAGE ON SCHEMA voting_app TO ${sql_user_name};
+    GRANT SELECT, UPDATE ON ALL TABLES IN SCHEMA voting_app TO ${sql_user_name};
+    INSERT INTO voting_app.choice (text, vote_count) VALUES('Tabs', 0);
+    INSERT INTO voting_app.choice (text, vote_count) VALUES('Spaces', 0);
+`;
+
+const deletion_script = "DROP SCHEMA IF EXISTS voting_app CASCADE";
+
+const postgresqlVotesTable = new Schema("postgresql-votes-schema", {
+    creatorName: sql_admin_name,
+    creatorPassword: sql_admin_password,
+    serverAddress: postgresqlRdsServer.address,
+    databaseName: postgresDatabase.name,
+    creationScript: creation_script,
+    deletionScript: deletion_script,
+    postgresUserName: postgresUser.name
+});
+```
+
+With the basic infrastructure and provider completed, we can write the application deployment code to ECS. We will use Pulumi Crosswalk, which is a collection of libraries that makes common infrastructure-as-code tasks in AWS easier and more secure by using well-architected best practices automatically.
+
+We'll first set up the server. The Network Listener is assigned the same port as the server, which is, in our case, port 5000. A set of environment variables representing our PostgreSQL connection credentials are passed directly to the `awsx.ecs.FargateService`. With AWS Crosswalk, what would have been over 150 lines of code is reduced to just under 20 lines of code.
+
+```typescript
+const serversideListener = new awsx.elasticloadbalancingv2.NetworkListener("server-side-listener", { port: 5000 });
+const serversideService = new awsx.ecs.FargateService("server-side-service", {
+    taskDefinitionArgs: {
+        containers: {
+            serversideService: {
+                image: awsx.ecs.Image.fromPath("server-side-service", "./serverside"),
+                memory: 512,
+                portMappings: [serversideListener],
+                environment: [
+                    { name: "USER_NAME", value: sql_user_name },
+                    { name: "USER_PASSWORD", value: sql_user_password },
+                    { name: "RDS_ADDRESS", value: postgresqlRdsServer.address },
+                    { name: "RDS_PORT", value: String(2000) },
+                    { name: "DATABASE_NAME", value: postgresDatabase.name },
+                ],
+            },
+        },
+    },
+});
+```
+
+The same is true for the client service, which can be reduced to a short and easy to understand format. By default, React uses port 3000, but it can set to a different port. The `SERVER_HOSTNAME` environment variable which we pass in is used when the container starts to generate a tiny configuration file at runtime called `serverParams.js` with the URL. This way, we do not have to rebuild the entire docker image should the server URL change.
```

Comment:
```suggestion
The same is true for the client service, which is reduced to a short and easy to understand format. By default, React uses port 3000, but it can set to a different port. The `SERVER_HOSTNAME` environment variable is passed in when the container starts to generate a tiny configuration file at runtime called `serverParams.js` with the URL. This way, we do not have to rebuild the entire docker image should the server URL change.
```